### PR TITLE
pkg/regionspan(ticdc): remove dead code and fix typos

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/4954

### What is changed and how it works?

- Remove dead code and fix typos
- reorder the code

#### Tests 

None

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/regionspan/span.go [hunk 1]
```diff
@@ -231,15 +224,15 @@ func IsSubSpan(sub ComparableSpan, parents ...ComparableSpan) bool {
 	return false
 }
 
-// ToComparableSpan returns a memcomparable span
+// ToComparableSpan returns a comparable span.
```
Comment (overvenus) [inline pkg/regionspan/span.go:227]: It's not a typo https://github.com/facebook/mysql-5.6/wiki/MyRocks-record-format

Comment (hi-rustin) [inline pkg/regionspan/span.go:227]:
```suggestion
// ToComparableSpan returns a memcomparable span.
// See: https://github.com/facebook/mysql-5.6/wiki/MyRocks-record-format
```

File: pkg/regionspan/span.go [hunk 2]
```diff
@@ -231,15 +224,15 @@ func IsSubSpan(sub ComparableSpan, parents ...ComparableSpan) bool {
...
 func ToComparableSpan(span Span) ComparableSpan {
 	return ComparableSpan{
 		Start: codec.EncodeBytes(nil, span.Start),
 		End:   codec.EncodeBytes(nil, span.End),
 	}
 }
 
-// ToComparableKey returns a memcomparable key.
+// ToComparableKey returns a comparable key.
```
Comment (hi-rustin) [inline pkg/regionspan/span.go:235]:
```suggestion
// ToComparableKey returns a memcomparable key.
```

File: pkg/regionspan/span.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/regionspan/span.go]: Thanks!

**Review by overvenus — APPROVED** [review]

**Pull request merged**